common/scripted_effects/wc_scheme_effects.txt

```diff
@@ -10,4 +10,357 @@
+		STUNNER = scope:owner
+	}
+
+	#Stress reduction
+	if = {
+		limit = {
+			scope:scheme.scheme_owner = {
+				has_trait = sadistic
+			}
+		}
+		scope:scheme.scheme_owner = {
+			stress_impact = {
+				sadistic = minor_stress_impact_loss
+			}
+		}
+	}
+	if = {
+		limit = {
+			exists = scope:targeted_rival
+		}
+		if = {
+			limit = {
+				scope:scheme.scheme_owner = {
+					has_trait = vengeful
+				}
+			}
+			scope:scheme.scheme_owner = {
+				stress_impact = {
+					vengeful = medium_stress_impact_loss
+				}
+			}
+		}
+		else_if = {
+			limit = {
+				scope:scheme.scheme_owner = {
+					has_trait = forgiving
+				}
+			}
+			scope:scheme.scheme_owner = {
+				stress_impact = {
+					base = medium_stress_impact_loss
+					forgiving = minor_stress_impact_gain
+				}
+			}
+		}
+		else = {
+			scope:scheme.scheme_owner = {
+				add_stress = medium_stress_loss
+			}
+		}
+	}
+	
+	scope:scheme = {
+		end_scheme = yes
+	}
+}
+
+
+stasis_trap_failure_effect = {
+	#Add Watchful Modifier to the target
+	scope:target = {
+		show_as_tooltip = {
+			add_character_modifier = {
+				modifier = watchful_modifier
+				days = watchful_modifier_duration
+			}
+		}
+	}
+
+	scope:target = {
+		trigger_event = stasis_trap_outcome.5002
+	}
+}
+
+successful_stasis_trap_outcome_event_option_effect = {
+	if = {
+		limit = {
+			faith = {
+				has_doctrine_parameter = piety_gain_from_successful_intrigue_schemes
+			}	
+		}
+		add_piety = minor_piety_gain
+	}
+}
+
+stun_character_effect = {
+	$TARGET$ = { save_temporary_scope_as = new_target }
+	$STUNNER$ = { save_temporary_scope_as = stunner }
+
+	scope:new_target = {
+		add_character_modifier = {
+			modifier = stasis_trap_stunned
+			years = 3
+		} 
+	}
+
+	hidden_effect = {
+		#Disband any scheme you have against them
+		scope:stunner = {
+			if = {
+				limit = {
+					any_scheme = {
+						scheme_target = scope:new_target
+					}
+				}
+				every_scheme = {
+					limit = {
+						scheme_target = scope:new_target
+					}
+					end_scheme = yes
+				}
+			}
+		}
+
+		#Send break up event if they are your lover
+		scope:stunner = {
+			if = {
+				limit = {
+					has_relation_lover = scope:new_target
+					NOT = { has_relation_rival = scope:new_target } #To enable really strange love stories
+				}
+				scope:new_target = {
+					trigger_event = {
+						id = lover.0101
+						days = { 14 30 }
+					}
+				}
+			}
+		}
+	}
+}
+
+dont_excecute_scheme_effect = {
+	scope:scheme = {
+		add_scheme_progress = -10 # Reset scheme progress
+	}
+
+	stress_impact = {
+		impatient = medium_stress_impact_gain
+		stubborn = medium_stress_impact_gain
+	}
+}
+
+abandon_scheme_effect = {
+	scope:scheme = {
+		end_scheme = yes
+	}
+
+	stress_impact = {
+		stubborn = medium_stress_impact_gain
+	}
+}
+
+restart_scheme_effect = {
+	play_music_cue = "mx_cue_stress"
+	custom_tooltip = restart_scheme_tt
+	
+	scope:scheme = {
+		add_scheme_progress = -10 # Reset scheme progress
+	}
+}
+
+hostile_scheme_opinion_effect = {
```

Comment:
So there's one problem and resolving will take a good amount of your time and I guess will be a good experience.

Basically, you can't pass `scope:scheme` to it because in many cases `scope:scheme` is over. Like in `stasis_trap_outcome.5001`.

Reply:
> Basically, you can't pass `scope:scheme` to it because in many cases `scope:scheme` is over. Like in `stasis_trap_outcome.5001`.

Here's the event.

Of course, I haven't murdered anyone.

Steps to reproduce:
- I replaced this part to make sure it's always discovered.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/117648731-a28d0780-b19f-11eb-8de6-8835bb01928d.png)

</details>

- Enabled `yesman` so everyone joins my plot.
- Invited many trolls.
- Waited 11 months until the attempt event fired.
- Saved (for future tests).
- Switched to my target.
- Got the event.

Reply:
I suppose you have to give up the idea of passing the scheme or make sure the scheme ends when all opinions are assigned.